### PR TITLE
Fixes #6392 Optimize WP Query in `rocket_url_to_postid()`

### LINKED ISSUE #6392: Review uses of WP_Query and consider using no_found_rows , update_post_term_cache and update_post_meta_cache to speed up processing and reduce SQL workload
**Context**
A user suggested the following through the Support: 

> Your query on line 245 of /inc/functions/posts.php can be made 5-10x faster with no_found_rows => true being added. Since you are no totals needed or pagination occurring you will skip the expensive SQL_CALC_FOUND_ROWS that query translates to.

Referring to [this](https://github.com/wp-media/wp-rocket/blob/dc721cd3cda91bae2d483345f6c231e32730a4ff/inc/functions/posts.php#L82C11-L82C31)

The use of WP Query implies a few steps on top of the query requested. More details [here](https://wordpress.stackexchange.com/questions/315369/is-no-found-rows-a-silver-bullet-for-wp-query-optimization-when-getting-all-po).

**To do**
Review our usage (at least the main ones) of WP Query and consider the use of the following option: no_found_rows, update_post_term_cache and update_post_meta_cache

### PR DESCRIPTION
**Pull request by remyperona**
## Description

Optimize the query by passing the args to prevent caching and found rows calculation.

Fixes #6392

## Type of change

- Enhancement (non-breaking change which improves an existing functionality).

## Is the solution different from the one proposed during the grooming?

No grooming, small change.

# Checklists

## Generic development checklist

- [x] My code follows the style guidelines of this project, with adapted comments and without new warnings.
- [x] The CI passes locally with my changes (including unit tests, integration tests, linter).

## Test summary

- [x] I triggered all changed lines of code at least once without new errors/warnings/notices.
- [ ] I validated all Acceptance Criteria of the related issues. (If applicable, provide proof).
- [ ] I validated all test plan the QA Review asked me to.

### CODE REVIEW

**Review by Miraeld — APPROVED** [review]

**Comment by Mai-Saad** [comment]
Related initial test plan https://wpmediaqa.testrail.io/index.php?/runs/view/816&group_by=cases:section_id&group_order=asc

File: inc/functions/posts.php [hunk 1]
```diff
@@ -250,7 +250,15 @@ function rocket_url_to_postid( string $url ) {
 				$query = (array) apply_filters( 'rocket_url_to_postid_query_args', $query, $url );
 
 				// Do the query.
-				$query = new WP_Query( $query );
+				$query = new WP_Query(
```
Comment (wordpressfan) [inline inc/functions/posts.php]: As far as I know, WP_Query constructor has only one argument, what I'm missing here?

**Review by wordpressfan — CHANGES_REQUESTED** [review]

File: inc/functions/posts.php (hunk 1 above)
Comment (wordpressfan) [reply inc/functions/posts.php]: @Tabrisrp I think we are fine now, just wanted to mention that those items will be outside the filter, and I think that's fine or u think to consider them into the filter so anyone can play with it if he faced a problem?

Comment (remyperona) [reply inc/functions/posts.php]: I think we don't want those to be overridden by the filter, as that would reduce optimization of the query.

**Review by wordpressfan — APPROVED** [review]

**Comment by hanna-meda** [comment]
Moved this to QA done for now without merging it until we check the 2 blocked tests in [here](https://wpmediaqa.testrail.io/index.php?/runs/view/816).

**Review by hanna-meda — APPROVED** [review]
[testrail-report-577.pdf](https://github.com/wp-media/wp-rocket/files/14448468/testrail-report-577.pdf)

**Pull request merged**